src/Momento.Sdk.Incubating/SimpleCacheClient.cs

```diff
@@ -169,7 +170,11 @@ public async Task<CacheDictionaryGetFieldResponse> DictionaryGetFieldAsync(strin
         }
         catch (ArgumentNullException e)
         {
-            return new CacheDictionaryGetFieldResponse.Error(new InvalidArgumentException(e.Message));
+            if (field == null)
+            {
+                return new CacheDictionaryGetFieldResponse.Error(Google.Protobuf.ByteString.Empty, new InvalidArgumentException(e.Message));
+            }
```

Comment:
When users pass null to `field`, I'm passing an empty ByteString to the error response type.
With the current `CacheDictionaryGetFieldResponse.Error` constructor, we need to pass ByteString, which works for both `cacheName` and `dictionaryName` are null but it would fail when `field` is null.
I feel like an empty ByteString is not the same as null and users could misinterpret the `field` value if they see an empty ByteString instead of null.
Is it better to make `CacheDictionaryGetFieldResponse.Error` accept null `field`? Or is it better to return an empty ByteString as `field` value?

@malandis Your input would be appreciated!

Reply:
Schwern wrote this a while back:

https://github.com/momentohq/standards-and-practices/blob/main/docs/client-specifications/error-handling.md#exceptions

It seems applicable here, IMO.  This seems like a case where if a user passes a `null` then it's a bug in the user's code and we should just fail fast and throw.  Then we would never have to worry about nulls inside of the Response class at all, which makes a lot more sense to me.

That said, I'm not sure how many more places in the .NET SDK this would impact; if it would be a big change then we should just make the `field` in the response object nullable like you were starting to do, Erika.  Especially if there are any APIs in the current 1.x releases of the .NET SDK that are doing it this way (wrapping null checks in Error response class); if that is true then we can't change it and we need to remain consistent in these methods.

In the event that we can't eliminate the possibility of nulls for the Error response object then you are right, Erika, an empty ByteString would be very confusing and null would be a better way to go.

I'm still not totally clear on how we are ending up with these nulls in the first place though, I guess the C# compiler must not actually be enforcing that you can't call our public methods with nulls even though we have expressed the types as non-nullable?

Reply:
@cprice404 
Got it.
Every data structure API has a check called `Utils.ArgumentNotNull` to make sure users are not passing any null values. If they do, then we return the error response type with the `ArgumentNullException`.

It looks like C# compiler only warns users if they are to pass a null value to a non-null parameter, and it's considered the build was successful:
```
/Users/erikatharp/client-sdk-dotnet-incubating/examples/DictionaryExample/Program.cs(36,20): warning CS8625: Cannot convert null literal to non-nullable reference type. [/Users/erikatharp/client-sdk-dotnet-incubating/examples/DictionaryExample/DictionaryExample.csproj]
  DictionaryExample -> /Users/erikatharp/client-sdk-dotnet-incubating/examples/DictionaryExample/bin/Debug/net6.0/DictionaryExample.dll

Build succeeded.

/Users/erikatharp/client-sdk-dotnet-incubating/examples/DictionaryExample/Program.cs(36,20): warning CS8625: Cannot convert null literal to non-nullable reference type. [/Users/erikatharp/client-sdk-dotnet-incubating/examples/DictionaryExample/DictionaryExample.csproj]
    1 Warning(s)
    0 Error(s)
```

So based on these, I'm going to go ahead and make `field` nullable for this PR.
But I guess ideally we want to throw the `ArgumentNullException` instead of returning the error response type for this case although there will be some work to modify the code.

Reply:
We do the null check for all the APIs in the main .NET SDK, so I will stick to the existing error response pattern (and yes! It should be nullable only for the error type)

Reply:
that sounds like the right plan then.  THanks.